Modules/Loadable/SubjectHierarchy/Widgets/qMRMLSubjectHierarchyComboBox.cxx

```diff
@@ -67,6 +69,7 @@ qMRMLSubjectHierarchyComboBoxPrivate::qMRMLSubjectHierarchyComboBoxPrivate(qMRML
   , MaximumNumberOfShownItems(20)
   , AlignPopupVertically(true)
   , TreeView(NULL)
+  , NoItemLabel(NULL)
```

Comment:
shouldn't we delete `TreeView` and `NoItemLabel` in a destructor for `qMRMLSubjectHierarchyComboBoxPrivate` ?

Reply:
> we delete TreeView (and now NoItemLabel) in a destructor for qMRMLSubjectHierarchyComboBoxPrivate

If the object  have parents, Qt takes care of memory management.

In case these object are orphan, they indeed need to be deleted. Can you check ?

> And delete the private class in qMRMLSubjectHierarchyComboBox destructor

Not required. The d pointer is a `QScopedPointer` and it already takes care of this.

See https://github.com/agirault/Slicer/blob/b620d00da3a3eeef699041622aff464507ede296/Modules/Loadable/SubjectHierarchy/Widgets/qMRMLSubjectHierarchyComboBox.h#L125

Reply:
Thanks for the explanation :+1: